internal/addrs/resource.go

```diff
@@ -393,10 +393,44 @@ func (r ConfigResource) Equal(o ConfigResource) bool {
 	return r.Module.Equal(o.Module) && r.Resource.Equal(o.Resource)
 }
 
+// Less returns true if the receiver should sort before the given other value
+// in a sorted list of addresses.
+func (r ConfigResource) Less(o ConfigResource) bool {
+	switch {
+
+	case len(r.Module) != len(o.Module):
+		return len(r.Module) < len(o.Module)
+
+	case r.Module.String() != o.Module.String():
+		return r.Module.Less(o.Module)
+
+	case r.Resource.Mode != o.Resource.Mode:
+		return r.Resource.Mode == DataResourceMode
+
+	case r.Resource.Type != o.Resource.Type:
+		return r.Resource.Type < o.Resource.Type
+
+	case r.Resource.Name != o.Resource.Name:
+		return r.Resource.Name < o.Resource.Name
+
+	default:
+		return false
+
+	}
+}
+
 func (r ConfigResource) configMoveableSigil() {
 	// AbsResource is moveable
 }
 
+type configResourceKey string
+
+func (k configResourceKey) uniqueKeySigil() {}
+
+func (r ConfigResource) UniqueKey() UniqueKey {
+	return configResourceKey(r.String())
+}
```

Comment:
How are these changes in the `addrs` package related to this PR?

Reply:
oh hmm good call out... this was a leftover from when I thought I was going to need to implement a new graph transformer, but I ended up not doing it and this is only here because I has already committed it. I'll rebase this commit out of the branch.